typing/mtype.ml

```diff
@@ -138,6 +138,17 @@ and strengthen_lazy_sig' ~aliasable sg p =
       sigelt :: strengthen_lazy_sig' ~aliasable rem p
   | (Sig_class_type _ as sigelt) :: rem ->
       sigelt :: strengthen_lazy_sig' ~aliasable rem p
+  | Sig_jkind(id, decl, vis) :: rem ->
+      let decl =
+        match decl.jkind_manifest with
+        | Some _ -> decl
+        | None ->
+          let manif =
+            Some (Jkind.Const.kconstr (Pdot(p, Ident.name id)))
+          in
+          { decl with jkind_manifest = manif }
+      in
+      Sig_jkind(id, decl, vis) :: strengthen_lazy_sig' ~aliasable rem p
```

Comment:
We can avoid an alloc in the case where there's already a manifest, like in `sig_make_manifest`.